File: src/012. Highly divisible triangular number/012.java
public class Main {

    public static void main(String[] args) {
        long triangle = 0;
        int i = 1;
        while (true) {
            triangle += i;
            if (numOfFactors(triangle) > 500) {
                System.out.println(triangle);
                break;
            }
            i++;
        }
    }
    
    public static int numOfFactors(long num) {
        int count = 0;
        long factor = 1;
        while (factor * factor < num)) {
            if (num % factor == 0) {
                count += 1;
            }
            factor += 1;
        }
        if (factor * factor > num) {
            return count * 2;
        } else {
            // perfect square
            return count * 2 + 1;
        }
    }
}

File: src/012. Highly divisible triangular number/012.py
def num_of_factors(num):
    n = 0
    factor = 1
    while factor * factor < num:
        if num % factor == 0:
            n += 1
        factor += 1
    if factor * factor > num:
        return n * 2
    else:
        # perfect square
        return n * 2 + 1

triangle = 0
i = 1
while True:
    triangle += i
    if num_of_factors(triangle) > 500:
        print(triangle)
        break
    i += 1
